### ISSUE TITLE
Add .net core as debug provider

### ISSUE THREAD

**Issue by cpunella** [Azure/vscode-kubernetes-tools#491]
Hi, 

With latest release nodejs has been added as debug provider. I was wondering when .net core will be supported as well.

Thanks

**Comment by itowlson** [comment]
We don't have this planned in at the moment.  Would you be interested in collaborating on implementing one?  If so we can definitely provide guidance and support; but not sure we have the resource to work on it ourselves at the moment I'm afraid.

cc @radu-matei

**Comment by pereiren** [comment]
Hi @itowlson, @ismael-soriano and me could be interested in collaborating on this. We want to debug our microservices using Minikube. Do you have any guide for starters?

**Comment by itowlson** [comment]
@pereiren @ismael-soriano That would be lovely - thank you heaps for the offer!

In terms of a starter guide, I'd suggest reviewing this PR which added the Node.js debug provider:

https://github.com/Azure/vscode-kubernetes-tools/pull/451/files

The key files of interest in that PR are `nodejsDebugProvider.ts` (where all the actual functionality sits) and `providerRegistry.ts` (which registers the debug provider).  The rest is I think mostly refactorings and configuration stuff.  Note this only shows attaching to a pod; for launching a pod under debug see the Java provider (`javaDebugProvider.ts`).  You may also find the Java provider useful as it may be probably closer to the .NET Core model in terms of launching the remote debugger - I haven't done enough with .NET Core to know, though, so don't take my word for it!

The methods in the debug provider are:

* `resolvePortsFromFile` - required for a _launch_ configuration, otherwise can return `undefined`
* `resolvePortsFromContainer` - required for an _attach_ configuration, otherwise can return `undefined`
* `isSupportedImage` - implement this if you want your provider automatically selected for Dockerfiles with certain base images, otherwise okay to return false
* `isDebuggerInstalled` - the Node provider assumes it is, because VS Code, but the Java one has an example of detecting a debugger extension and trying to download it if not present
* `getDebuggerType` - I _think_ this is only to identify the provider in the UI
* `startDebugging` - should set up the appropriate debugging configuration object and call `vscode.debug.startDebugging`

I hope that's enough to get you on the right track.  I have to admit I'm not the expert on the debugging subsystem - so if you have questions, I may have to figure it out alongside you, which is good learning for me but please be forgiving if I'm slow or make mistakes!  Good luck and many thanks!

**Comment by pereiren** [comment]
@itowlson Thank you for the information! We will analyze during this sprint (next 2 weeks) how much effort we need for it and if everything fits with our company goals we would define all the tasks required for it. 

I'll keep you updated! Best regards!

**Comment by liguobao** [comment]
Looking forward to it, thanks!

**Comment by pereiren** [comment]
> Looking forward to it, thanks!

@liguobao Hi, unfortunately I moved to another company and we didn't have time to invest on this. Sorry :( 

In any case, you can do it using vsdbg and then attaching the code to the running process. The only thing you cannot debug is the startup process of the microservice (if I am not wrong)

**Comment by patrickcarnahan** [comment]
I might want to pick this work up, or someone from my team anyway. Still something that is interesting for a new feature?

**Comment by itowlson** [comment]
@patrickcarnahan Yes, very much of interest.  Thanks!

**Comment by patrickcarnahan** [comment]
@itowlson we will very likely be happy with just the attach functionality and have no immediate need for launch. Is a version trimmed down in this manner acceptable as a standalone PR?

**Comment by itowlson** [comment]
@patrickcarnahan Yes, absolutely, as long as it fails gracefully if the user tries to use it in launch mode.

**Comment by tedchamb** [comment]
I added PR #674, to support dotnet debugging in the attach scenario,

**Comment by patrickcarnahan** [comment]
@itowlson @tedchamb is a co-worker of mine who pursued the fix we discussed previously. Definitely open for feedback!
